src/p2p/net_node.inl

```diff
@@ -1370,67 +1489,46 @@ bool node_server<t_payload_net_handler>::make_new_connection_from_peerlist(
                     if (is_addr_recently_failed(pe.adr))
                         return true;
 
-                    const bool have_net16 =
```

Comment:
Is the `have_net16` check dubious and not useful? I guess the earlier intention was to preferably avoid connecting to someone within the same network if there's an option not to digging back into the git blame.

Reply:
The new way of finding a peers just wasn't using it. So was doing work that would be later ignored.

It relied on the exponential function selecting later items in the list very infrequently. Where the new `select_best_peer` does a separate calculation.

We could probably add another penalty to the peer scoring if we have already got a node on the same network, but im not convinced its worth it